### PR TITLE
Issue 6383

### PR DESCRIPTION
**Pull request by MadWasp79**
Added default units of measurement #6383

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 1]
```diff
@@ -674,6 +682,60 @@ private String getFormattedInt(String value) {
...
+						+ " "+ mapActivity.getResources().getString(R.string.yard);
+				} else {
+					value = value + " " + mapActivity.getResources().getString(R.string.m);
+				}
+				break;
+			case "distance":
+				if (metricSystem == OsmandSettings.MetricsConstants.MILES_AND_FEET ||
+						metricSystem == OsmandSettings.MetricsConstants.MILES_AND_YARDS ||
+						metricSystem == OsmandSettings.MetricsConstants.MILES_AND_METERS) {
+					value = String.valueOf(df.format(Double.valueOf(value)*1000/OsmAndFormatter.METERS_IN_ONE_MILE))
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: do Double.valueOf(value) * 1000 once only, use spaces between operands
btw - we have getFormattedDistance, use it

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 2]
```diff
@@ -674,6 +682,60 @@ private String getFormattedInt(String value) {
 		}
 	}
 
+	private String[] getFormattedPrefixAndText(String key, String prefix, String value) {
+		DecimalFormat df = new DecimalFormat("#.##");
+		df.setRoundingMode(RoundingMode.CEILING);
+		switch (key){
+			case "width":
+			case "height":
+				prefix = Algorithms.capitalizeFirstLetter(key);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: will it be translated to other langs?
use Algorithms.capitalizeFirstLetter for locale-insensitive text only

File: OsmAnd/res/values/strings.xml [hunk 3]
```diff
@@ -2423,6 +2423,8 @@
     <string name="km">km</string>
     <string name="km_h">km/h</string>
     <string name="m">m</string>
+    <string name="cubic_m">m³</string>
+    <string name ="capacity">capacity</string>
```
Comment (crimean) [inline OsmAnd/res/values/strings.xml]: Should start from capital letter

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 4]
```diff
@@ -674,6 +682,60 @@ private String getFormattedInt(String value) {
...
+						metricSystem == OsmandSettings.MetricsConstants.MILES_AND_METERS) {
+					value = String.valueOf(df.format(Double.valueOf(value)*1000/OsmAndFormatter.METERS_IN_ONE_MILE))
+						+ " " + mapActivity.getResources().getString(R.string.mile);
+				} else if (metricSystem == OsmandSettings.MetricsConstants.NAUTICAL_MILES) {
+					value = String.valueOf(df.format(Double.valueOf(value)*1000/OsmAndFormatter.METERS_IN_ONE_NAUTICALMILE))
+						+ " " + mapActivity.getResources().getString(R.string.nm);
+				} else {
+					value = value + " " + mapActivity.getResources().getString(R.string.km);
+				}
+				prefix = Algorithms.capitalizeFirstLetter(key);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: do not redefine method's params, create new

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 5]
```diff
@@ -674,6 +682,60 @@ private String getFormattedInt(String value) {
...
+			case "spots":
+			case "seats":
+				prefix = prefixConstructor(prefix, mapActivity.getResources().getString(R.string.capacity));
+				break;
+		}
+
+		return new String[]{prefix, value};
+	}
+
+	private String prefixConstructor(String prefix, String units){
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: if method will not be used anymore - do not create it for one call

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java (hunk 2 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: fixed

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: Changed to getFormattedDistance for distance but retained local conversion for height/length/width.

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 6]
```diff
@@ -685,55 +686,54 @@ private String getFormattedInt(String value) {
...
 				break;
 			case "maxweight":
-				value = value + " " +  mapActivity.getResources().getString(R.string.cubic_m);
+				formattedValue = value + " " + mapActivity.getResources().getString(R.string.metric_ton);
 				break;
 			case "students":
 			case "spots":
 			case "seats":
-				prefix = prefixConstructor(prefix, mapActivity.getResources().getString(R.string.capacity));
+				prefix = (!prefix.isEmpty()) ? (prefix + ", " + mapActivity.getResources().getString(R.string.capacity))
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: done

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 7]
```diff
@@ -685,55 +686,54 @@ private String getFormattedInt(String value) {
...
 			case "height":
-				prefix = Algorithms.capitalizeFirstLetter(key);
+				if (key.equals("width")) {
+					prefix = mapActivity.getResources().getString(R.string.width_label);
+				} else {
+					prefix = mapActivity.getResources().getString(R.string.height_label);
+				}
 			case "depth":
 			case "seamark_height":
+				double valueAsDouble = Double.valueOf(value);
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: don't need that much roundness on smaller numbers

File: OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java [hunk 8]
```diff
@@ -685,55 +686,54 @@ private String getFormattedInt(String value) {
...
 		DecimalFormat df = new DecimalFormat("#.##");
 		df.setRoundingMode(RoundingMode.CEILING);
-		switch (key){
+		String formattedValue = "";
+		switch (key) {
 			case "width":
 			case "height":
-				prefix = Algorithms.capitalizeFirstLetter(key);
+				if (key.equals("width")) {
+					prefix = mapActivity.getResources().getString(R.string.width_label);
```
[the remark this answers is not in the document]
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java]: done

**Pull request merged**